libvast/src/detail/process.cpp

```diff
@@ -24,39 +25,37 @@ namespace vast::detail {
 
 namespace {
 
-caf::optional<vast::path> objectpath_dynamic(const void* addr) {
+caf::expected<path> objectpath_dynamic(const void* addr) {
   Dl_info info;
   if (!dladdr(addr, &info))
-    return caf::none;
+    return make_error(ec::unspecified, "failed to execute dladdr()");
   if (!info.dli_fname)
-    return caf::none;
+    return make_error(ec::unspecified, "addr not in an mmapped region");
   return info.dli_fname;
 }
 
-caf::optional<vast::path> objectpath_static() {
-#if __linux__
+caf::expected<path> objectpath_static() {
+#ifdef VAST_LINUX
   struct stat sb;
   auto self = "/proc/self/exe";
   if (lstat(self, &sb) == -1)
-    return caf::none;
+    return make_error(ec::unspecified, "lstat() returned with error");
   auto size = sb.st_size ? sb.st_size + 1 : PATH_MAX;
   std::vector<char> buf(size);
-  if (readlink(self, buf.data(), size) == -1) {
-    return caf::none;
-  }
+  if (readlink(self, buf.data(), size) == -1)
+    return make_error(ec::unspecified, "readlink() returned with error");
   return path{buf.data()};
 #else
-  return caf::none;
+  return make_error(ec::unimplemented, "");
```

Comment:
Good call to improve the specificity of the error code. However, either just return the error code or add a non-empty string, but `""` adds more noise than benefit.

Reply:
I just assumed the second argument was mandatory.